src/Campaigns/Routes/PublishCampaign.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Give\Campaigns\Routes;
+
+use Give\API\RestRoute;
+use Give\Campaigns\Models\Campaign;
+use Give\Campaigns\Routes\Traits\RestResponses;
+use Give\Campaigns\ValueObjects\CampaignStatus;
+use Give\Framework\Exceptions\Primitives\Exception;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class PublishCampaign implements RestRoute
+{
+    use RestResponses;
+
+    /** @var string */
+    protected $endpoint = 'campaigns/publish/(?P<id>[0-9]+)';
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute()
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => 'PUT',
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'string',
+                        'required' => true,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function handleRequest(WP_REST_Request $request): WP_Rest_Response
+    {
+        $campaignId = $request->get_param('id');
+        $campaign = Campaign::find($campaignId);
+
+        if ( ! $campaign) {
+            return $this->notFoundResponse(esc_html__(sprintf('Campaign %s not found.', $campaignId), 'give'));
+        }
+
+        try {
+            $campaign->status = CampaignStatus::ACTIVE();
+            $campaign->save();
+
+            $response = json_encode($campaign->toArray());
+        } catch (Exception $e) {
+            return $this->badRequestResponse($e);
```

Comment:
We may not have to catch the `Exception` to handle the response type. If I remember correctly, the WP_REST_API will convert an `Exception` to a `500` error response.

Reply:
Resolved in 7f6062c3bcc0f7f47453fe7a1cc851f14cace14d and 7f64906a2d650e4af1d6f95a6ba444ff62404ec4 and ea7a3d193d985b101ce808c5311ba7e3cd16e0ee